google/cloud/datastore/helpers.py

```diff
@@ -47,14 +48,12 @@ def _get_meaning(value_pb, is_list=False):
               means it just returns a list of meanings. If all the
               list meanings agree, it just condenses them.
     """
-    meaning = None
     if is_list:
 
-        values = (
-            value_pb._pb.array_value.values
-            if hasattr(value_pb, "_pb")
-            else value_pb.array_value.values
-        )
+        # Use 'raw' protobuf message if possible
+        value_pb = getattr(value_pb, "_pb", value_pb)
```

Comment:
Changes like this (there are a few in the PR) seem be a pure refactor of the old implementation, unless I am missing something that further helps performance?

Reply:
Three-arg `getattr` does the test in C, versus in Python's own branch machinery.  At least in some cases, it can be a speedup.